src/plugins/compressors/sz_threadsafe_plugin.cc

```diff
@@ -0,0 +1,424 @@
+    set(options, "sz_threadsafe:lossless_compressor", threadsafe_params->losslessCompressor);
+    set(options, "sz_threadsafe:sample_distance", threadsafe_params->sampleDistance);
+    set(options, "sz_threadsafe:pred_threshold", threadsafe_params->predThreshold);
+    set(options, "sz_threadsafe:sz_mode", threadsafe_params->szMode);
+    set(options, "sz_threadsafe:gzip_mode", threadsafe_params->gzipMode);
+    set(options, "sz_threadsafe:error_bound_mode", threadsafe_params->errorBoundMode);
+    set_type(options, "sz_threadsafe:error_bound_mode_str", pressio_option_charptr_type);
+	  set(options, "sz_threadsafe:abs_err_bound", threadsafe_params->absErrBound);
+	  set(options, "sz_threadsafe:rel_err_bound", threadsafe_params->relBoundRatio);
+	  set(options, "sz_threadsafe:psnr_err_bound", threadsafe_params->psnr);
+	  set(options, "sz_threadsafe:pw_rel_err_bound", threadsafe_params->pw_relBoundRatio);
+	  set(options, "sz_threadsafe:segment_size", threadsafe_params->segment_size);
+	  set(options, "sz_threadsafe:snapshot_cmpr_step", threadsafe_params->snapshotCmprStep);
+	  set(options, "sz_threadsafe:accelerate_pw_rel_compression", threadsafe_params->accelerate_pw_rel_compression);
+	  set_type(options, "sz_threadsafe:prediction_mode", pressio_option_int32_type);
+    set_type(options, "sz_threadsafe:data_type", pressio_option_double_type);
+#if PRESSIO_SZ_VERSION_GREATEREQ(2,1,11,1)
+    set(options, "sz_threadsafe:exafel:peaks_segs", exafel_peaks_segs);
+    set(options, "sz_threadsafe:exafel:peaks_rows", exafel_peaks_rows);
+    set(options, "sz_threadsafe:exafel:peaks_cols", exafel_peaks_cols);
+    set(options, "sz_threadsafe:exafel:num_peaks", static_cast<unsigned int>(exafel_params.numPeaks));
+    set(options, "sz_threadsafe:exafel:calib_panel", exafel_calibPanel);
+    set(options, "sz_threadsafe:exafel:tolerance", exafel_params.tolerance);
+    set(options, "sz_threadsafe:exafel:bin_size", static_cast<unsigned int>(exafel_params.binSize));
+    set(options, "sz_threadsafe:exafel:sz_dim", static_cast<unsigned int>(exafel_params.szDim));
+    set(options, "sz_threadsafe:exafel:peak_size", static_cast<unsigned int>(exafel_params.peakSize));
+#endif
+    set(options, "sz_threadsafe:app", app.c_str());
+#ifdef HAVE_RANDOMACCESS
+    set(options, "sz_threadsafe:random_access", threadsafe_params->randomAccess);
+#endif
+    set(options, "sz_threadsafe:user_params", user_params);
+    return options;
+  }
+
+  int set_options_impl(struct pressio_options const& options) override {
+
+    struct sz_params* sz_threadsafe_param;
+    std::string config_file;
+    if(get(options, "sz_threadsafe:config_file", &config_file) == pressio_options_key_set) {
+      SZ_Finalize();
+      SZ_Init(config_file.c_str());
+    } else if (get(options, "sz_threadsafe:config_struct", (void**)&sz_threadsafe_param) == pressio_options_key_set) {
+      SZ_Finalize();
+      SZ_Init_Params(sz_threadsafe_param);
```

Comment:
This isn't right either, we want to copy the new params that the user passed us into our config structure.